### PR TITLE
Don't try to delete from tx table if it's not in use.

### PR DESCRIPTION
**Pull request by mtrippled**
### Type of Change

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Refactor (non-breaking change that only restructures code)
- [ ] Tests (You added tests for code that already exists, or your new feature included in this PR)
- [ ] Documentation Updates
- [ ] Release

### CODE REVIEW

File: src/ripple/app/ledger/impl/LedgerMaster.cpp [hunk 1]
```diff
@@ -2344,9 +2344,9 @@ LedgerMaster::getFetchPackCacheSize() const
 boost::optional<LedgerIndex>
 LedgerMaster::minSqlSeq()
 {
-#ifdef RIPPLED_REPORTING
     if (app_.config().reporting())
     {
+#ifdef RIPPLED_REPORTING
```
Comment (miguelportilla) [inline src/ripple/app/ledger/impl/LedgerMaster.cpp:2349]: Do we report or log an error if reporting is enabled through configuration and the build does not support it?

Comment (cjcobb23) [reply src/ripple/app/ledger/impl/LedgerMaster.cpp:2349]: It actually only throws an error in this situation if you are also using Cassandra as a backend. Should probably change that. The best place to detect this and throw would be in either the `ReportingETL` constructor or `ReportingETL::run()`

**Review by cjcobb23 — APPROVED** [review]
Nice catch!

File: src/ripple/app/misc/SHAMapStoreImp.cpp [hunk 2]
```diff
@@ -695,6 +696,9 @@ SHAMapStoreImp::clearPrior(LedgerIndex lastRotated)
     if (health())
         return;
 
+    if (! app_.config().useTxTables())
```
Comment (miguelportilla) [inline src/ripple/app/misc/SHAMapStoreImp.cpp:699]: Can also be
```if (!transactionDb_)``` or ```if (transactionDb == nullptr)```

**Review by miguelportilla — APPROVED** [review]
Left some minor nits but not holding back as is. 👍

**Comment by cjcobb23** [comment]
@mtrippled Here is a commit that you can cherry pick to fix the compilation issue: https://github.com/cjcobb23/rippled/commit/f111f61e898dabc97024b3f1cc8933ba42c793b6

It also adds some additional checks. It checks that the code as built with `-Dreporting=1` if trying to run in reporting mode, and also checks that `use_tx_tables=1` (or is not specified) when running in reporting mode, since reporting mode always writes to the transaction tables.

**Comment by ximinez** [comment]
@mtrippled Another compilation issue in certain MSVC builds is fixed by this commit: https://github.com/ximinez/rippled/commit/07fc0425a713f77e0644dc89564e39f28c885cd6

**Pull request closed without merging**